isobus/include/isobus/isobus/can_stack_logger.hpp

```diff
@@ -37,36 +38,98 @@ namespace isobus
 		};
 
 		/// @brief The constructor for a CANStackLogger
-		CANStackLogger();
+		CANStackLogger() = default;
 
 		/// @brief The destructor for a CANStackLogger
-		~CANStackLogger();
+		~CANStackLogger() = default;
 
 		/// @brief Gets called from the CAN stack to log information. Wraps sink_CAN_stack_log.
 		/// @param[in] level The log level for this text
 		/// @param[in] logText The text to be logged
 		static void CAN_stack_log(LoggingLevel level, const std::string &logText);
 
+		/// @brief Gets called from the CAN stack to log information. Wraps sink_CAN_stack_log.
+		/// @param[in] level The log level for this text
+		/// @param[in] format A format string of text to log, similar to printf
+		/// @param[in] args A list of printf style arguments to use with the format string when logging
+		template<typename... Args>
+		static void CAN_stack_log(LoggingLevel level, const std::string &format, Args... args)
+		{
+			int size_s = std::snprintf(nullptr, 0, format.c_str(), args...) + 1; // Extra space for '\0'
+			if (size_s > 0)
+			{
+				auto size = static_cast<std::size_t>(size_s);
+				std::unique_ptr<char[]> buf(new char[size]);
+				std::snprintf(buf.get(), size, format.c_str(), args...);
+				CAN_stack_log(level, std::string(buf.get(), buf.get() + size - 1)); // We don't want the '\0' inside
+			}
```

Comment:
What if `snprintf` has a formatting error, e.g. it returns a negative number? We might want to print the string without formatting it so the log message is still preserved? Could be really frustrating to not see a log message show up I feel like

Reply:
Yep, good point, I've added another condition to at least print the format string if snprintf returns a negative number.